### PR TITLE
Multiple assignment for `UPDATE SET`

### PR DESCRIPTION
**Pull request by nickgerrets**
Solves #7069 

Implemented Transform for the `T_MultiAssignRef` expression type.

A ParserException is thrown when the left side of the assignment doesn't have an equal amount of columns as the right side of the assignent, i.e:
`UPDATE tbl SET (a, b, c) = (1, 2);`

### CODE REVIEW

File: test/sql/update/test_multiple_assignment.test [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# name: test/sql/update/test_multiple_assignment.test
+# description: Update with multiple assignment
+# group: [update]
+
```
Comment (Tishj) [inline test/sql/update/test_multiple_assignment.test:4]: Can we add a test with empty lists?
such as:
`() = (1, 2)`
`(1,2) = ()`

File: src/parser/transform/expression/transform_multi_assign_reference.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		idx_t idx = 1;
+		auto list = func.args->head;
+		while (list && idx < static_cast<idx_t>(root.colno)) {
+			list = list->next;
+			++idx;
+		}
+
+		// Not enough columns (ie. (x, y, z) = (1, 2) )
+		if (!list) {
+			throw ParserException("Too few columns for Multiple Assignment");
```
Comment (Tishj) [inline src/parser/transform/expression/transform_multi_assign_reference.cpp:33]: Can we provide a more insightful error message here?
"Could not perform multiple assignment, target expects %d values, only %d were provided"

File: src/parser/transform/expression/transform_multi_assign_reference.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		// Explicitly only allow ROW function
+		char const *function_name =
+		    PGPointerCast<duckdb_libpgquery::PGValue>(func.funcname->tail->data.ptr_value)->val.str;
+		if (function_name == nullptr || strlen(function_name) != 3 || strncmp(function_name, "row", 3) != 0) {
+			return TransformExpression(root.source);
+		}
+
+		// Too many columns (ie. (x, y) = (1, 2, 3) )
+		if (root.ncolumns < func.args->length) {
+			throw ParserException("Too many columns for Multiple Assignment");
```
Comment (Tishj) [inline src/parser/transform/expression/transform_multi_assign_reference.cpp:20]: Can this be more insightful as well, maybe something like:
"Could not perform multiple assignment, expected only %d values, found %d instead"

File: src/parser/transform/expression/transform_multi_assign_reference.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		while (list && idx < static_cast<idx_t>(root.colno)) {
+			list = list->next;
+			++idx;
+		}
+
+		// Not enough columns (ie. (x, y, z) = (1, 2) )
+		if (!list) {
+			throw ParserException("Too few columns for Multiple Assignment");
+		}
+		return TransformExpression(reinterpret_cast<duckdb_libpgquery::PGNode *>(list->data.ptr_value));
```
Comment (Tishj) [inline src/parser/transform/expression/transform_multi_assign_reference.cpp:35]: Unrelated comment:
Without looking at the grammar this is super confusing, as I would expect `MultiAssignRef` to contain the full list of assignments: `(a, b, c) = (1,2,3)` but instead we create a `MultiAssignRef` node for every value in the left hand side: `a`, `b` and `c`

**Review by Tishj — CHANGES_REQUESTED** [review]
Thanks for the PR, the changes here look great, I just had some requests for more tests and improving some error messages.

**Review by Tishj — APPROVED** [review]
Thanks for the changes!

**Comment by Mytherin** [comment]
Thanks! Looks great!

**Pull request merged**